proxy_map.go

```diff
@@ -121,6 +122,35 @@ func (m *Map) NewLockContext(ctx context.Context) context.Context {
 	return context.WithValue(ctx, lockIDKey{}, lockID(m.refIDGen.NextID()))
 }
 
+type MapListener struct {
+	EntryAdded       func(event *EntryNotified)
+	EntryRemoved     func(event *EntryNotified)
+	EntryUpdated     func(event *EntryNotified)
+	EntryEvicted     func(event *EntryNotified)
+	EntryExpired     func(event *EntryNotified)
+	EntryAllEvicted  func(event *EntryNotified)
+	EntryAllCleared  func(event *EntryNotified)
+	EntryMerged      func(event *EntryNotified)
+	EntryInvalidated func(event *EntryNotified)
+	EntryLoaded      func(event *EntryNotified)
+}
+
+func (m *Map) AddListener(ctx context.Context, listener MapListener, includeValue bool) (types.UUID, error) {
+	return types.NewUUID(), errors.New("s")
```

Comment:
re: `errors.New("s")` is that a placeholder for a proper error?

Reply:
Yes, The implementation is not done as you can clearly see.